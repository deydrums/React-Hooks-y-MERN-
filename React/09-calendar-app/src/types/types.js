export const types = {
    uiOpenModal: '[ui] Open modal',
    uiCloseModal: '[ui] Close modal',

    eventAddnew: '[event] Add new',
    eventSetActive: '[event] Set active',
    eventUnsetActive: '[event] Unset active',
}