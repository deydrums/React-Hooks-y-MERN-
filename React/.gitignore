# proyectos
/00-proyectos