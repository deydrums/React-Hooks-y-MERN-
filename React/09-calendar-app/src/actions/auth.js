import { fetchWithoutToken } from "../helpers/fetch"
import { types } from "../types/types";
import Swal from 'sweetalert2';



export const startLogin = (email, password) => {
    return async(dispatch) => {
        const resp = await fetchWithoutToken('auth',{email,password},'POST');
        const body = await resp.json();
        console.log(body);
        if(resp.ok) {
            localStorage.setItem('token',body.token);
            localStorage.setItem('token-init-date', new Date().getTime());
            dispatch(login({
                uid: body.uid,
                name: body.name
            }));
        }else{
            Swal.fire('Error',body.message?body.message:'Ha ocurrido un error','error');
        }
    }
}

export const startRegister = (name, email, password) => {
    return async(dispatch) => {
        const resp = await fetchWithoutToken('auth/new',{name,email,password},'POST');
        const body = await resp.json();
        console.log(body);
        if(resp.ok) {
            localStorage.setItem('token',body.token);
            localStorage.setItem('token-init-date', new Date().getTime());
            dispatch(login({
                uid: body.uid,
                name: body.name
            }));
        }else{
            Swal.fire('Error',body.message?body.message:'Ha ocurrido un error','error');
        }
    }
}

const login = (user) => ({
    type: types.authLogin,
    payload: user
});