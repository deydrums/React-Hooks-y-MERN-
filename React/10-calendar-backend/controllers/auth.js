const {response} = require('express');
const User = require('../models/User');
var bcrypt = require('bcryptjs');
const {generateJWT} = require('../helpers/jwt');

const createUser = async(req,res = response)=>{
    const {email, password} = req.body;
    try {
        let user = await User.findOne({email: email});
        if(user){
            return res.status(400).json({
                ok: false,
                message: 'El correo ya fue registrado anteriormente'
            });
        };
        user = new User(req.body);
        //Encriptar contraseÃ±a
        const salt = bcrypt.genSaltSync();
        user.password = bcrypt.hashSync(password,salt);

        await user.save();
        //Generar nuestro JWT
        const token = await generateJWT(user.id, user.name);

    
        res.status(201).json({
            ok: true,
            message: 'Registro de usuario exitoso',
            uid: user.id,
            name: user.name,
            token
        });

    } catch (error) {
        console.log(error);
        res.status(500).json({
            ok: false,
            message: 'Ha ocurrido un error, intenta de nuevo'
        })
    }
}

const loginUser = async(req,res = response)=>{
    const {email, password} = req.body

    try {
        const user = await User.findOne({email: email});
        if(!user){
            return res.status(404).json({
                ok: false,
                message: 'Usuario no encontrado'
            });
        };

        //Confirmar los passwords
        const validPassword = bcrypt.compareSync(password, user.password);
        if(!validPassword){
            return res.status(400).json({
                ok: false,
                message: 'Password incorrecto'
            });
        };

        //Generar nuestro JWT
        const token = await generateJWT(user.id, user.name);

        res.status(200).json({
            ok: true,
            message: 'Login correcto',
            uid: user.id,
            name: user.name,
            token
        });


    } catch (error) {
        console.log(error);
        res.status(500).json({
            ok: false,
            message: 'Ha ocurrido un error, intenta de nuevo'
        });
    }
}

const renewToken = async(req,res = response)=>{
    const uid = req.uid;
    const name = req.name;

    //Generar nuestro JWT
    const token = await generateJWT(uid, name);

    res.json({
        ok: true,
        message: 'Nuevo token generado',
        uid,
        name,
        token
    })
}

module.exports = {
    createUser,
    loginUser,
    renewToken
};