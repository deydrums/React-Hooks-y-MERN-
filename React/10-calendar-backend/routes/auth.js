/**
 * 
 * Routas de Usuarios / Auth
 * host + /api/auth
 *  
 * */ 

const {Router} = require('express');
const { check } = require('express-validator');
const router = Router();
const { createUser, loginUser, renewToken } = require('../controllers/auth');
const { paramsValidator } = require('../middlewares/params-validator');
const { validateJWT } = require('../middlewares/validate-jwt');

router.post(
    '/new',
    [
        check('name', 'El nombre no es valido').not().isEmpty(),
        check('email', 'El email no es valido').isEmail(),
        check('password', 'El password debe de ser de 6 caracteres').isLength({min: 6}), 
        paramsValidator
    ],
    createUser
    );

router.post(
    '/',
    [
        check('email', 'El email no es valido').isEmail(),
        check('password', 'El password debe de poseer minimo 6 caracteres').isLength({min: 6}),
        paramsValidator
    ],
     loginUser);

router.get('/renew', validateJWT, renewToken);

module.exports = router;