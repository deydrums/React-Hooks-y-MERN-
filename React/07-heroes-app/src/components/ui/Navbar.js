import React from 'react'
import { Link, NavLink } from 'react-router-dom'

export const Navbar = () => {
    return (
            <nav className="navbar navbar-expand-lg navbar-dark bg-dark">
                <div className="container-fluid">
                    <Link 
                        className="navbar-brand" 
                        to="/"
                    >
                        Asociaciones
                    </Link>

                    <button className="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbarSupportedContent"
                    aria-controls="navbarSupportedContent" aria-expanded="false" aria-label="Toggle navigation">
                    <span className="navbar-toggler-icon"></span>
                    </button>

                    <div className="collapse navbar-collapse" id="navbarSupportedContent">
                        <div className="navbar-nav me-auto mb-2 mb-lg-0">
                            <NavLink 
                                activeClassName="active"
                                className="nav-item nav-link" 
                                exact
                                to="/marvel"
                            >
                                Marvel
                            </NavLink>

                            <NavLink 
                                activeClassName="active"
                                className="nav-item nav-link" 
                                exact
                                to="/dc"
                            >
                                DC
                            </NavLink>
                        </div>

                        <div className="navbar-nav navbar-right">
                        
                            <NavLink 
                            activeClassName="active"
                            className="nav-item nav-link" 
                            exact
                            to="/login"
                            >
                                Logout
                            </NavLink>
                        </div>
                    </div>
                </div>
            </nav>

    )
}