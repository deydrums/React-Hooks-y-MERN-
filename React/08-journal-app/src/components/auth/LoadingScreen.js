import React from 'react'

export const LoadingScreen = () => {
    return (
        <div>
            <div className="auth__main">
                <div className="auth__loading">
                <div class="page-loader">
                    <div></div>
                    <div></div>
                    <div></div>
                </div>
                </div>
            </div>
        </div>
    )
}
