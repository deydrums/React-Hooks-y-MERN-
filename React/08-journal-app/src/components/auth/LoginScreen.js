import React from 'react';
import {Link} from 'react-router-dom';
import { useForm } from '../../hooks/useForm';
import { useDispatch } from 'react-redux';
import { login } from '../../actions/auth';

export const LoginScreen = () => {

    const dispatch = useDispatch();

    const [formValues, handleInputChange, reset] = useForm({
        email: 'admin@admin.com',
        password: '123456'
    });

    const {email, password} = formValues;

    const handleLogin = (e) => {
        e.preventDefault();
        dispatch(login(12345,'David'));
    }

    return (
        <>
            <h3 className="auth__title">Iniciar sesión</h3>
            <form onSubmit={handleLogin}>
                <input
                    type="text"
                    placeholder="Email..."
                    name = "email"
                    className="auth__input"
                    autoComplete="off"
                    value = {email}
                    onChange={handleInputChange}
                />
                <input
                    type="password"
                    placeholder="Password..."
                    name = "password"
                    className="auth__input"
                    value = {password}
                    onChange={handleInputChange}
                />
                <button
                    type="submit"
                    className = "btn btn-primary btn-block"
                >
                    Entrar
                </button>
                
                <div className="auth__social-networks">
                    <p>Iniciar con redes sociales</p>
                    <div 
                        className="google-btn"
                    >
                        <div className="google-icon-wrapper">
                            <img className="google-icon" src="https://upload.wikimedia.org/wikipedia/commons/5/53/Google_%22G%22_Logo.svg" alt="google button" />
                        </div>
                        <p className="btn-text">
                            <b>Entrar con Google</b>
                        </p>
                    </div>
                </div>
                <Link 
                    to="/auth/register"
                    className="link"
                >
                    Registrarse aquí
                </Link>
            </form>
        </>
    )
}
