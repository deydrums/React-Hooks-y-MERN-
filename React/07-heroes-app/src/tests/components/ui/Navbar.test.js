import {mount} from 'enzyme';
import React from 'react';
import '@testing-library/jest-dom';
import { MemoryRouter, Router } from 'react-router-dom';
import { AuthContext } from '../../../auth/AuthContext';
import { Navbar } from '../../../components/ui/Navbar';
import {types} from '../../../types/types';

describe('Pruebas en Navbar', () => {

    const historyMock = {
        push: jest.fn(),
        replace: jest.fn(),
        location: {},
        listen: jest.fn(),
        createHref: jest.fn()
    }

    const contextValue = {
        dispatch: jest.fn(),
        user: {
            logged: false,
            name: 'David',
            email: 'dagarcia100@gmail.com'
        }
    }
    
    const wrapper = mount(
        <MemoryRouter>
            <AuthContext.Provider value={contextValue}>
                <Router history={historyMock}>
                    <Navbar/>
                </Router>
            </AuthContext.Provider>
        </MemoryRouter>
    );

    afterEach(() => {
        jest.clearAllMocks();
    });

    test('Debe de mostrarse correctamente ', () => {
        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('.text-info').text().trim()).toBe(contextValue.user.email);
    });

    test('Debe de llamar el logout y usar el history', () => {
        wrapper.find('.logout').prop('onClick')();
        expect(contextValue.dispatch).toHaveBeenCalledTimes(1); 
        expect(contextValue.dispatch).toHaveBeenCalledWith({
            type: types.logout
        });
        expect(historyMock.replace).toHaveBeenCalledTimes(1);
        expect(historyMock.replace).toHaveBeenCalledWith('/login');
    });

    
})
