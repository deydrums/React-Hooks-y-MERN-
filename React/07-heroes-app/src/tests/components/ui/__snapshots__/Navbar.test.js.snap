// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en Navbar Debe de mostrarse correctamente  1`] = `
<nav
  className="navbar navbar-expand-lg navbar-dark bg-dark"
>
  <div
    className="container-fluid"
  >
    <img
      alt="logo"
      className="App-logo navbar-logo"
      src="logo.svg"
    />
    <a
      className="navbar-brand"
      onClick={[Function]}
    >
      HeroesApp
    </a>
    <button
      aria-controls="navbarSupportedContent"
      aria-expanded="false"
      aria-label="Toggle navigation"
      className="navbar-toggler"
      data-bs-target="#navbarSupportedContent"
      data-bs-toggle="collapse"
      type="button"
    >
      <span
        className="navbar-toggler-icon"
      />
    </button>
    <div
      className="collapse navbar-collapse"
      id="navbarSupportedContent"
    >
      <div
        className="navbar-nav me-auto mb-2 mb-lg-0"
      >
        <a
          aria-current={null}
          className="nav-item nav-link"
          onClick={[Function]}
        >
          Marvel
        </a>
        <a
          aria-current={null}
          className="nav-item nav-link"
          onClick={[Function]}
        >
          DC
        </a>
        <a
          aria-current={null}
          className="nav-item nav-link"
          onClick={[Function]}
        >
          Buscar
        </a>
      </div>
      <div
        className="navbar-nav navbar-right"
      >
        <span
          className="nav-item nav-link text-info"
        >
          dagarcia100@gmail.com
        </span>
        <a
          className="nav-item nav-link logout"
          onClick={[Function]}
        >
          Logout
        </a>
      </div>
    </div>
  </div>
</nav>
`;
